src/main/java/com/google/devtools/build/lib/remote/disk/CombinedDiskHttpBlobStore.java

```diff
@@ -39,17 +40,21 @@
 
   private final SimpleBlobStore remoteCache;
   private final OnDiskBlobStore diskCache;
+  private final RemoteOptions options;
 
-  public CombinedDiskHttpBlobStore(OnDiskBlobStore diskCache, SimpleBlobStore remoteCache) {
+  public CombinedDiskHttpBlobStore(OnDiskBlobStore diskCache, SimpleBlobStore remoteCache, RemoteOptions options) {
     this.diskCache = Preconditions.checkNotNull(diskCache);
     this.remoteCache = Preconditions.checkNotNull(remoteCache);
+    this.options = options;
   }
 
   @Override
   public void putActionResult(ActionKey actionKey, ActionResult actionResult)
       throws IOException, InterruptedException {
     diskCache.putActionResult(actionKey, actionResult);
-    remoteCache.putActionResult(actionKey, actionResult);
+    if (!options.incompatibleUploadLocalResultsDiskOnly || options.remoteUploadLocalResults) {
```

Comment:
I find the following version a bit more readable, but it's up to you: to change it or not :)
 `if (options.incompatibleUploadLocalResultsDiskOnly && !options.remoteUploadLocalResults)`

Reply:
I agree with @ishikhman here; not a strong preference, but it seems a bit cleaner

Reply:
You'd need to negate the entire clause for those to be equivalent: 
`!(options.incompatibleUploadLocalResultsDiskOnly && !options.remoteUploadLocalResults)`

I think either way works, but I prefer the original : )

Reply:
I agree with @SrodriguezO, also prefer the original. More suggestions are welcome.